overlord/snapstate/snapmgr.go

```diff
@@ -31,19 +31,21 @@ import (
 
 // Install initiates a change installing snap.
 // Note that the state must be locked by the caller.
-func Install(change *state.Change, snap, channel string) error {
+func Install(change *state.Change, snap, channel string, flags snappy.InstallFlags) error {
 	t := change.NewTask("install-snap", fmt.Sprintf(i18n.G("Installing %q"), snap))
 	t.Set("name", snap)
 	t.Set("channel", channel)
+	t.Set("flags", flags)
```

Comment:
It seems easier to cook this as:
```
type installState struct {
        Name string `json:"name"`
        ...
}
```
instead of having to marshal/unmarshal one by one.